starlite_jwt/middleware.py

```diff
@@ -73,3 +73,93 @@ async def authenticate_request(self, connection: "HTTPConnection") -> Authentica
             raise NotAuthorizedException()
 
         return AuthenticationResult(user=user, auth=token)
+
+
+class JWTCookieAuthenticationMiddleware(JWTAuthenticationMiddleware):
+    """JWT Cookie Authentication Middleware.
+
+    Args:
+        JWTCookieAuthenticationMiddleware (_type_): _description_
+    """
+
+    def __init__(  # pylint: disable=[too-many-arguments]
+        self,
+        algorithm: str,
+        app: "ASGIApp",
+        auth_header: str,
+        auth_cookie: str,
+        auth_cookie_options: CookieOptions,
+        retrieve_user_handler: "Callable[[str], Awaitable[Any]]",
+        token_secret: str,
+        exclude: Optional[Union[str, List[str]]],
+    ):
+        """This Class is a Starlite compatible JWT authentication middleware.
+
+        It checks incoming requests for an encoded token in the auth header specified,
+        and if present retrieves the user from persistence using the provided function.
+
+        Args:
+            app: An ASGIApp, this value is the next ASGI handler to call in the middleware stack.
+            retrieve_user_handler: A function that receives an instance of 'Token' and returns a user, which can be
+                any arbitrary value.
+            token_secret: Secret for decoding the JWT token. This value should be equivalent to the secret used to encode it.
+            auth_header: Request header key from which to retrieve the token. E.g. 'Authorization' or 'X-Api-Key'.
+            auth_cookie: Cookie name from which to retrieve the token. E.g. 'token' or 'accessToken'.
+            algorithm: JWT hashing algorithm to use.
+            exclude: A pattern or list of patterns to skip.
+        """
+        super().__init__(
+            algorithm=algorithm,
+            app=app,
+            auth_header=auth_header,
+            retrieve_user_handler=retrieve_user_handler,
+            token_secret=token_secret,
+            exclude=exclude,
+        )
+
+        self.auth_cookie = auth_cookie
+        self.auth_cookie_options = auth_cookie_options or CookieOptions()
+
+    async def authenticate_request(self, connection: "HTTPConnection") -> AuthenticationResult:
+        """Given an HTTP Connection, parse the JWT api key stored in the header
+        and retrieve the user correlating to the token from the DB.
+
+        Args:
+            connection: An Starlette HTTPConnection instance.
+
+        Returns:
+            AuthenticationResult
+
+        Raises:
+            [NotAuthorizedException][starlite.exceptions.NotAuthorizedException]: If token is invalid or user is not found.
+        """
+        encoded_token = coalesce([connection.headers.get(self.auth_header), connection.cookies.get(self.auth_cookie)])
```

Comment:
you dont need a utility for this, i think this bit is over-engineered:

```suggestion
        encoded_token = connection.headers.get(self.auth_header) or connection.cookies.get(self.auth_cookie)
```